hotsos/core/ycheck/engine/properties/requires/types/snap.py

```diff
@@ -22,19 +22,61 @@ def installed_revisions(self):
         _revisions = []
         for p in self.installed:
             _revisions.append(self.packaging_helper.get_revision(p))
+
         return _revisions
 
-    def package_revision_within_ranges(self, pkg, ranges):
+    @cached_property
+    def installed_channels(self):
+        _channels = []
+        for p in self.installed:
+            _channels.append(self.packaging_helper.get_channel(p))
+
+        return _channels
+
+    def package_info_matches(self, pkg, pkg_info):
+        """
+        If snap package has versions and/or channels we check them here.
+        """
         result = False
-        revision = int(self.packaging_helper.get_revision(pkg))
-        # compare revisions for the time being
-        for item in sorted(ranges, key=lambda i: i['max'],
-                           reverse=True):
-            r_max = int(item['max'])
-            r_min = int(item['min'])
-            if r_min <= revision <= r_max:
-                result = True
-        return result
+
+        # Ranges must be specified as min:max
+        if any((i.get('min') is None and i.get('max') is not None) or
+               (i.get('max') is None and i.get('min') is not None)
+                for i in pkg_info):
+            raise Exception("ranges for package '{}' contain one or more "
+                            "range that do not specify both a min and a max "
```

Comment:
nit: ranges, or values?

Reply:
yeah i dunno whats best, it is a list of ranges so ...